### DISCUSSION TITLE
CVE-2024-21742

### DISCUSSION THREAD

**Discussion by Alex-Kidston** [quarkusio/quarkus#39672] [Community]
Hi - [CVE-2024-21742](https://nvd.nist.gov/vuln/detail/CVE-2024-21742) is outstanding for the apache-mime4j-core dependency to move to v0.8.10 - is there a targeted fix version? thanks in advance...

**Comment by sberyozkin** [comment]
@Alex-Kidston Hi, I believe Quarkus does not use MIME4J DOM anywhere

**Comment by Alex-Kidston** [comment]
@sberyozkin unfortunately, I can see the apache-mime4j v0.8.9 dependency here:
[quarkus-bom](https://github.com/quarkusio/quarkus/blob/main/bom/application/pom.xml)
I think that's where our mend scanning is finding it...

**Reply by sberyozkin** [reply]
@Alex-Kidston We'll probably need to silence the various scanners with the version bump, but Quarkus itself does not use that code AFAIK

**Comment by Alex-Kidston** [comment]
@sberyozkin completely understand...yes, mend is pretty unforgiving...if a dependency is mentioned, it's evaluated for CVE's...thanks for investigating!

**Comment by Alex-Kidston** [comment]
Resolved by this [PR](https://github.com/quarkusio/quarkus/pull/39679)

**Comment by Alex-Kidston** [comment]
@sberyozkin I hate to say it, but even after pulling in 3.9.1, we're seeing the apache-mime4j-core v0.8.9 as a dependency - here's the gradle scan output:
```
compileClasspath
0.058s
org.apache.james:apache-mime4j-core:0.8.9
quarkusDevBaseRuntimeClasspathConfiguration
0.016s
org.apache.james:apache-mime4j-core:0.8.9
quarkusProdBaseRuntimeClasspathConfiguration
0.254s
org.apache.james:apache-mime4j-core:0.8.9
quarkusProdRuntimeClasspathConfiguration
0.048s
org.apache.james:apache-mime4j-core:0.8.9
quarkusProdRuntimeClasspathConfigurationDeployment
0.697s
org.apache.james:apache-mime4j-core:0.8.9
runtimeClasspath
0.041s
org.apache.james:apache-mime4j-core:0.8.9
```
All of our build.gradle quarkus dpendencies are set to 3.9.1 - is there something we're missing? thanks as ever for your help...

**Reply by sberyozkin** [reply]
@Alex-Kidston It is managed at the bom level, so you probably need to wait for the corresponding 3.9.1 platform release, @aloubyansky, is it correct ?

**Reply by Alex-Kidston** [reply]
@sberyozkin looks like that's already happened - [PR](https://github.com/quarkusio/quarkus-platform/pull/1173)

**Reply by sberyozkin** [reply]
@aloubyansky, @gsmet Can you comment please why the mime4j version managed at the Quarkus BOM level and set in Quarkus 3.9.1 to 0.8.11, is still 0.8.9 at the Quarkus platform level ?

**Reply by gsmet** [reply]
I see it in the core BOM but I don't see it in the Platform BOM, which is a bit weird.

We will need @aloubyansky to have a look at what's going on.

**Reply by aloubyansky** [reply]
If it's not in the platform BOM it means the version constraint is invalid and was filtered out.
```
<dependency>
  <groupId>org.apache.james</groupId>
  <artifactId>apache-mime4j</artifactId>
  <version>0.8.11</version>
</dependency>
```
implies it's a JAR artifact. There is no such JAR artifact though. https://repo1.maven.org/maven2/org/apache/james/apache-mime4j/0.8.11/

**Reply by sberyozkin** [reply]
Thanks @aloubyansky @gsmet, I should've checked it.
https://repo1.maven.org/maven2/org/apache/james/apache-mime4j/0.8.9/ does not have it either though, and in 0.8.10, inside the zip file they have apache-mime4j-dom-0.8.10.jar. So I'm not sure what should be done but can revert #39679 if you'd like.

**Reply by aloubyansky** [reply]
The following `org.apache.james` artifacts are found among our dependencies:
```
↳ org.apache.james:apache-mime4j-dom:0.8.9
↳ org.apache.james:apache-mime4j-storage:0.8.9
↳ org.apache.james:apache-mime4j-core:0.8.9
```
We could add all of the above to the BOM and remove `apache-mime4`.

**Reply by sberyozkin** [reply]
@aloubyansky Thanks, I can try to handle it during the next few days to fix my earlier PR, definitely before the next 3.9.x

**Comment by Alex-Kidston** [comment]
@sberyozkin issue resolved on our end, whitesource scan is clean - thank you for your help!

**Comment by sberyozkin** [comment]
Thanks @Alex-Kidston, and @aloubyansky for recommending the right fix
